src/library/scala/concurrent/Future.scala

```diff
@@ -531,62 +581,61 @@ object Future {
-    override def zip[U](that: Future[U]): Future[(Nothing, U)] = this
-    override def zipWith[U, R](that: Future[U])(f: (Nothing, U) => R)(implicit executor: ExecutionContext): Future[R] = this
-    override def fallbackTo[U >: Nothing](that: Future[U]): Future[U] = this
-    override def mapTo[S](implicit tag: ClassTag[S]): Future[S] = this
-    override def andThen[U](pf: PartialFunction[Try[Nothing], U])(implicit executor: ExecutionContext): Future[Nothing] = this
-
-    override def toString: String = "Future(<never>)"
+    override final def onComplete[U](f: Try[Nothing] => U)(implicit executor: ExecutionContext): Unit = ()
+    override final def isCompleted: Boolean = false
+    override final def value: Option[Try[Nothing]] = None
+    override final def failed: Future[Throwable] = this
+    override final def foreach[U](f: Nothing => U)(implicit executor: ExecutionContext): Unit = ()
+    override final def transform[S](s: Nothing => S, f: Throwable => Throwable)(implicit executor: ExecutionContext): Future[S] = this
+    override final def transform[S](f: Try[Nothing] => Try[S])(implicit executor: ExecutionContext): Future[S] = this
+    override final def transformWith[S](f: Try[Nothing] => Future[S])(implicit executor: ExecutionContext): Future[S] = this
+    override final def map[S](f: Nothing => S)(implicit executor: ExecutionContext): Future[S] = this
+    override final def flatMap[S](f: Nothing => Future[S])(implicit executor: ExecutionContext): Future[S] = this
+    override final def flatten[S](implicit ev: Nothing <:< Future[S]): Future[S] = this
+    override final def filter(p: Nothing => Boolean)(implicit executor: ExecutionContext): Future[Nothing] = this
+    override final def collect[S](pf: PartialFunction[Nothing, S])(implicit executor: ExecutionContext): Future[S] = this
+    override final def recover[U >: Nothing](pf: PartialFunction[Throwable, U])(implicit executor: ExecutionContext): Future[U] = this
+    override final def recoverWith[U >: Nothing](pf: PartialFunction[Throwable, Future[U]])(implicit executor: ExecutionContext): Future[U] = this
+    override final def zip[U](that: Future[U]): Future[(Nothing, U)] = this
+    override final def zipWith[U, R](that: Future[U])(f: (Nothing, U) => R)(implicit executor: ExecutionContext): Future[R] = this
+    override final def fallbackTo[U >: Nothing](that: Future[U]): Future[U] = this
+    override final def mapTo[S](implicit tag: ClassTag[S]): Future[S] = this
+    override final def andThen[U](pf: PartialFunction[Try[Nothing], U])(implicit executor: ExecutionContext): Future[Nothing] = this
+    override final def toString: String = "Future(<never>)"
   }
 
-  /** A Future which is always completed with the Unit value.
+  /** A Future which is completed with the Unit value.
    */
-  val unit: Future[Unit] = successful(())
+  final val unit: Future[Unit] = fromTry(successOfUnit)
```

Comment:
It's not obvious to me what the motivation for this change is. I don't object to it particularly, but it looks perhaps slightly less readable, and it's only executed once regardless.

Reply:
We're reusing the same allocation to represent the successOfUnit.

Reply:
Aha, I missed the other places it was used, but I see now